cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/media/class-filter.php

```diff
@@ -522,17 +522,99 @@ public function filter_video_embeds( $html, $id, $attachment ) {
 	 */
 	public function pre_filter_rest_content( $response, $post, $request ) {
 		$context = $request->get_param( 'context' );
-		if ( 'edit' === $context ) {
-			$data                   = $response->get_data();
-			$content                = $data['content']['raw'];
-			$data['content']['raw'] = $this->filter_out_local( $content );
 
-			$response->set_data( $data );
+		if ( 'edit' !== $context ) {
+			return $response;
 		}
 
+		$data    = $response->get_data();
+		$content = $this->classic_image_to_block( $data['content']['raw'] );
+		$content = $this->video_shortcode_to_block( $content );
+
+		$data['content']['raw'] = $this->filter_out_local( $content );
+		$response->set_data( $data );
+
 		return $response;
 	}
 
+	/**
+	 * Converts video shortcode to a Gutenberg block.
+	 *
+	 * @param string $content
+	 * 
+	 * @return string
+	 */
+	protected function video_shortcode_to_block( $content ) {
+		if ( ! has_shortcode( $content, 'video' ) ) {
+			return $content;
+		}
+
+		$shortcodes = $attrs = array();
+
+		preg_match_all( '/' . get_shortcode_regex() . '/', $content, $shortcodes );
+
+		for ( $i = 0; $i < count( $shortcodes[3] ); $i++ ) {
+			preg_match_all( '/(mp4|id)="([^"]*)"/i', $shortcodes[3][$i], $attrs );
+	
+			// @TODO: Find a way to generate a video block
+			$video = '<!-- wp:video {"id":' . $attrs[2][1] . '} --><figure class="wp-block-video"><video autoplay src="' . $attrs[2][0] . '"></video></figure><!-- /wp:video -->';
+	
+			$content = str_replace( [ '<p>' . $shortcodes[0][$i] . '</p>', $shortcodes[0][$i] ], $video, $content );
+		}
+
+		return $content;
+	}
+
+	/**
+	 * Converts classic image to a Gutenberg block.
+	 *
+	 * @param string $content
+	 * 
+	 * @return string
+	 */
+	protected function classic_image_to_block( $content ) {
+		if ( strpos( $content, '<img class=' ) === false ) {
+			return $content;
+		}
+
+		$dom = new \DOMDocument();                
+		@$dom->loadHTML( $content, LIBXML_HTML_NODEFDTD );
```

Comment:
@dugajean 

I am wondering why we would want to silence the errors/warnings here.
Did you run into any while working on this?

Reply:
`loadHTML` complains when the HTML you try to load isn't a full-fledged HTML document (ie. no <html>, <body>, etc.). So a way to remedy this is by silencing it as the warnings aren't really saying anything important.

Reply:
More info here: https://stackoverflow.com/questions/9149180/domdocumentloadhtml-error

Reply:
thanks for the explanation and comment in the code.